### PR TITLE
ci: add push and pull workflows

### PR DESCRIPTION
**Pull request by teran-joseluis**
#### Resolved task/US: Setup CI Workflows for push and pulls actions.
* * * *

## What's in this pull request?

This PR refers to CI, having two .yml files for our workflow, which has the next jobs; `build` and `test` to automatize the project, and also `check_commits_convention`, `check_branches_convention` for every **push** to **main, develop, release and test** branches and `semantic_pull_request` and `check_commit_messages` for each **PR** that will be opened. Besides of `caching` Gradle dependencies in branches, which helps to avoid downloading and recompiling dependencies every time the pipeline runs, which significantly speeds up the build process.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [x] Continuous integration events (configuring actions)

## Checklist:

- [ ] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have checked my code and corrected any misspellings

### CODE REVIEW

**Comment by teran-joseluis** [comment]
I just saw that the test work did not pass, so @zamoranoalicia wouldn't it be good to change the **dsl** of the project to **Groovy** in its entirety instead of using it together with **Kotlin**?

I ask this because most of the errors I had when running the project locally end up referring to Kotlin and IDE configurations. Since, our project we are working on refers entirely to **Java**, so I think we could just use **Groovy**.

**Comment by zrock-dev** [comment]
I believe the reason this [pipeline](https://github.com/SoftDevSix/Argos/actions/runs/11472950137?pr=5) is failing because there is a mistake in the workflow file
image
The test job should also used java 17 but is using java 11
image

**Comment by teran-joseluis** [comment]
@zrock-dev yeah, you were right. Thank you so much.

File: .github/workflows/check-push.yml
Comment (zrock-dev) [inline .github/workflows/check-push.yml:1]: Each push to the listed branches would be through a PR, and the PR workflow is already building and testing. I don't think would be resource wise to have this workflow.

File: .github/workflows/check-pull-request.yml [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+        uses: actions/setup-java@v4
+        with:
+          java-version: '17'
+          distribution: 'adopt'
+          cache: 'gradle'
+
+      - name: Run Tests with Gradle
+        run: ./gradlew test --no-daemon
+
+  semantic_pull_request:
```
Comment (zrock-dev) [inline .github/workflows/check-pull-request.yml:50]: How does this job adds value to the pipeline?

File: .github/workflows/check-pull-request.yml [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+          subjectPattern: ^(?![A-Z]).+$
+          subjectPatternError: |
+            The subject "{subject}" found in the pull request title "{title}"
+            didn't match the configured pattern. Please ensure that the subject
+            doesn't start with an uppercase character.
+          headerPattern: '^(\w*)(?:\(([\w$.\-*/ ]*)\))?: (.*)$'
+          headerPatternCorrespondence: type, scope, subject
+          validateSingleCommit: true
+
+  check_commit_messages:
```
Comment (zrock-dev) [inline .github/workflows/check-pull-request.yml:82]: how does this job adds value to the pipeline?

File: .github/workflows/check-pull-request.yml [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+        uses: actions/setup-java@v4
+        with:
+          java-version: '17'
+          distribution: 'adopt'
+          cache: 'gradle'
+
+      - name: Build with Gradle
+        run: ./gradlew build --no-daemon
+
+  test:
```
Comment (zrock-dev) [inline .github/workflows/check-pull-request.yml:29]: sonarqube is handling the testing.

**Review by zrock-dev — CHANGES_REQUESTED** [review]
Good job, I have requested some changes

**Review by zamoranoalicia — APPROVED** [review]
LGTM

File: .github/workflows/check-push.yml
Comment (teran-joseluis) [reply .github/workflows/check-push.yml:1]: We need to ensure each PR has building and testing state passed and not wait to push events to verify it.

File: .github/workflows/check-pull-request.yml (hunk 1 above)
Comment (teran-joseluis) [reply .github/workflows/check-pull-request.yml:50]: It helps us to maintain a convention PR title during its opening state so we have good descriptive PR titles.

File: .github/workflows/check-pull-request.yml (hunk 2 above)
Comment (teran-joseluis) [reply .github/workflows/check-pull-request.yml:82]: It helps us to maintain a convention during commiting a file or set according its type.

File: .github/workflows/check-pull-request.yml (hunk 3 above)
Comment (teran-joseluis) [reply .github/workflows/check-pull-request.yml:29]: Yeah, but we also need to ensure that each PR had all the tests passed and to wait SonarQube to verify it.

**Pull request closed without merging**